### ISSUE TITLE
Remove libjansson dependency

### ISSUE THREAD

**Issue by majjejjam** [fejfighter/pgtk-emacs-flatpak#16]
It seems [libjansson has been replaced by a custom parser](https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=a5df4d92e37a176396577ac901b85025a6952376), so we should be able to drop it as a dependency, right?

**Comment by fejfighter** [comment]
Yes we absolutely can drop it as a dependency now.

I had made the changes locally last week when it was merged.
I'm having separate build issues with appstream content which has made me hesitant to push

**Comment by majjejjam** [comment]
I actually recently [forked this project ](https://codeberg.org/majjejjam/pgtk-emacs-flatpak) to add some opinionated patches (rounded corners and titlebar syncing colors with rest the rest of the frame). I also added support for SDK Extensions which I thought would probably be appropriate here, but didn't find time to create an issue or submit a PR yet.

Sorry, what is appstream? And what's difficult to differentiate?

**Comment by fejfighter** [comment]
Do the SDK extensions still need to get passed in/defined at launch? that was the main reason I avoided using it.

Appstream is the meta data info for flathub and gnome software type things. I can't seem to opt out of it.
the issue this change will introduce is that the app ID is now the same as the flathub version (`org.gnu.emacs`), rather than `org.gnu.emacs-pgtk`

**Comment by majjejjam** [comment]
> Do the SDK extensions still need to get passed in/defined at launch? that was the main reason I avoided using it.

Yes. You need to add extensions to an environmental variable: `FLATPAK_ENABLE_SDK_EXT=rust-nightly,texlive,llvm17` as an example. I do this with the help of [Flatseal](https://flathub.org/sv/apps/com.github.tchx84.Flatseal). Adding or removing requires restarting Emacs for the changes to take effect.

> Appstream is the meta data info for flathub and gnome software type things. I can't seem to opt out of it.
the issue this change will introduce is that the app ID is now the same as the flathub version (org.gnu.emacs), rather than org.gnu.emacs-pgtk

Thanks for the explanation! I take it _you_ have the standard Emacs flatpak installed so there's a conflict? My fork uses capitalization, so the app id is `org.gnu.Emacs`. Another solution could be to use your own domain (something like `org.fejfighter.emacs`) or just keep the pgtk suffix.

**Comment by fapdash** [comment]
@majjejjam @fejfighter From what I understand libjanson was not removed from Emacs, Géza Herman contributed a json parser, so json -> lisp is handled by the new code. lisp -> json will still use libjansson if available.

See https://mail.gnu.org/archive/html/emacs-devel/2024-03/msg00299.html
